kolibri/plugins/facility/assets/src/views/users/UsersRootPage/index.vue

```diff
@@ -279,6 +279,12 @@
         fetchClasses();
       });
 
+      watch([numAppliedFilters, searchTerm], (newValues, oldValues) => {
+        if (newValues[0] !== oldValues[0] || newValues[1] !== oldValues[1]) {
+          clearSelectedUsers();
+        }
+      });
```

Comment:
The logic here looks good to me. However the behavior needs to also apply to the NewUserPage and UsersTrashPage components.

You could look at putting this logic into the [useUserManagement.js](https://github.com/learningequality/kolibri/blob/3938485b54e46975c17760d5734a40414316db7e/kolibri/plugins/facility/assets/src/composables/useUserManagement.js#L16) composable - looks like we have the `search` and `numAppliedFilters` and it also houses the list of `selectedUsers` - so might be able to feed 3 birds with 1 stone since the other user table pages use useUserManagement for these values.